service/notifications.go

```diff
@@ -67,16 +68,29 @@ type notificationMessage struct {
 	Context     string                   `json:"context"`
 	Events      []notificationEvent      `json:"events"`
 	Recipients  []notificationRecipients `json:"recipients"`
+	ID          string                   `json:"id"`
 }
 
-func (producer *AvailabilityStatusNotifier) EmitAvailabilityStatusNotification(id *identity.Identity, emailNotificationInfo *m.EmailNotificationInfo) error {
+func (producer *AvailabilityStatusNotifier) EmitAvailabilityStatusNotification(id *identity.Identity, emailNotificationInfo *m.EmailNotificationInfo, guidPrefix string) error {
 	writer, err := kafka.GetWriter(&conf.KafkaBrokerConfig, notificationTopic)
 	if err != nil {
 		return fmt.Errorf(`could not get Kafka writer to emit an availability status notification: %w`, err)
 	}
 
 	defer kafka.CloseWriter(writer, "emit availability status notification")
 
+	notificationMessageGuid := uuid.New().String()
+	if guidPrefix != "" {
+		notificationMessageGuid = fmt.Sprintf("%s-%s", guidPrefix, notificationMessageGuid)
+	}
+
+	l.Log.Infof("[tenant_id: %s][source_id: %s] Publishing notification status message, status %s changed from: %s to %s",
+		emailNotificationInfo.TenantID,
+		emailNotificationInfo.SourceID,
+		emailNotificationInfo.PreviousAvailabilityStatus,
+		emailNotificationInfo.CurrentAvailabilityStatus,
+		notificationMessageGuid)
```

Comment:
I think this log message is a bit confusing, it is saying `Publishing notification status message, available changed from: unavailable status-listener-adsfadsfafds`, maybe we could put the guid as part of the log fields? `l.Log.Withfield("notification-guid", notificationMessageGuid).Infof(...)`? Just a thought.

Reply:
yes it was confusing I rephrase it and added `notification-guid` field - I also added to other log message in this function.

Log message now:

```json
{
   "@timestamp":"2022-08-16T11:33:30.649Z",
   "@version":1,
   "app":"source-api-go",
   "caller":"github.com/RedHatInsights/sources-api-go/service.(*AvailabilityStatusNotifier).EmitAvailabilityStatusNotification",
   "filename":"/Users/liborpichler/Projects/sources-api-go/service/notifications.go:87",
   "hostname":"lpichler1-mac",
   "labels":{
      "app":"source-api-go"
   },
   "level":"info",
   "message":"[tenant_id: 2][source_id: 51] Publishing notification status message, status changed from 'available' to 'unavailable'",
   "notification-guid":"notifier-middleware-5d152285-84d9-423e-b1b6-707d029649ed",
   "tags":[
      "source-api-go"
   ]
}

```

thanks!